metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/sql/DirectSqlSavePartition.java

```diff
@@ -418,6 +446,10 @@ public void addUpdateDropPartitions(final QualifiedName tableQName, final Table
             "INSERT INTO PARTITION_KEY_VALS(PART_ID,PART_KEY_VAL,INTEGER_IDX) VALUES (?,?,?)";
         static final String PARTITION_KEY_VALS_DELETES =
             "DELETE FROM PARTITION_KEY_VALS WHERE PART_ID in (%s)";
+        static final String PARTITIONS_SELECT_ALL =
+            "SELECT P.PART_ID, P.SD_ID, S.SERDE_ID FROM DBS D JOIN TBLS T ON D.DB_ID=T.DB_ID "
+                + "JOIN PARTITIONS P ON T.TBL_ID=P.TBL_ID JOIN SDS S ON P.SD_ID=S.SD_ID "
+                + "WHERE D.NAME=? and T.TBL_NAME=? limit %d";
         static final String PARTITIONS_SELECT =
```

Comment:
This two are almost same except the last part. I would factor them together.

Reply:
Wanted to keep it separate to make it explicit rather than have one with more OR logic.